styx-common/src/test/java/com/spotify/styx/storage/BigTableStorageTest.java

```diff
@@ -113,7 +118,34 @@ public void shouldReturnExecutionDataForWorkflow() throws Exception {
                    .get(0), is(ExecStatus.create(Instant.ofEpochMilli(4L), "SUBMITTED")));
     assertThat(workflowInstanceExecutionData.get(1).triggers().get(0).executions().get(0).statuses()
                    .get(1), is(ExecStatus.create(Instant.ofEpochMilli(5L), "STARTED")));
+  }
 
+  @Test
+  public void shouldPaginateExecutionDataForWorkflow() throws Exception {
+    setUp(0);
+    storage.writeEvent(SequenceEvent.create(Event.triggerExecution(WFI1, "triggerId1"), 0L, 0L));
+    storage.writeEvent(SequenceEvent.create(Event.triggerExecution(WFI2, "triggerId2"), 0L, 3L));
+    storage.writeEvent(SequenceEvent.create(Event.triggerExecution(WFI3, "triggerId3"), 0L, 3L));
+
+    List<WorkflowInstanceExecutionData> workflowInstanceExecutionData =
+        storage.executionData(WORKFLOW_ID1, WFI2.parameter(), 100);
```

Comment:
Ah! So the `offset` parameter is inclusive. This is unexpected to me and not how pagination works to me. You would have to guess the next ID to not get duplicates between pages.